featurebyte/query_graph/node/metadata/config.py

```diff
@@ -26,15 +27,146 @@ class BaseCodeGenConfig(BaseModel):
 
 class OnDemandViewCodeGenConfig(BaseCodeGenConfig):
     """
-    OnDemandViewCodeGenConfig is used to control the code generating style.
+    OnDemandViewCodeGenConfig is used to control the code generating style for Feast on demand feature view.
 
     input_df_name: str
         Input dataframe name
     """
 
-    input_df_name: str
-    output_df_name: str
-    on_demand_function_name: str
+    input_df_name: str = Field(default="inputs")
+    output_df_name: str = Field(default="df")
+    on_demand_function_name: str = Field(default="on_demand_feature_view")
+
+
+class OnDemandFunctionCodeGenConfig(BaseCodeGenConfig):
+    """
+    OnDemandFunctionCodeGenConfig is used to control the code generating style for DataBricks on demand function.
+
+    input_var_prefix: str
+        Input variable prefix for the function
+    request_input_var_prefix: str
+        Input variable prefix for the request column input
+    output_dtype: DBVarType
+        Output variable data type
+    """
+
+    sql_function_name: str = Field(default="odf_func")
+    sql_input_var_prefix: str = Field(default="x")
+    sql_request_input_var_prefix: str = Field(default="r")
+    sql_comment: str = Field(default="")
+    function_name: str = Field(default="on_demand_feature_function")
+    input_var_prefix: str = Field(default="col")
+    request_input_var_prefix: str = Field(default="request_col")
+    output_dtype: DBVarType
+
+    @classmethod
+    def to_py_type(cls, dtype: DBVarType) -> str:
+        """
+        Convert DBVarType to Python type
+
+        Parameters
+        ----------
+        dtype: DBVarType
+            Internal DBVarType
+
+        Returns
+        -------
+        str
+            Python type
+
+        Raises
+        ------
+        ValueError
+            If the input dtype is not supported
+        """
+        output = dtype.to_type_str()
+        if output:
+            return output
+        if dtype in DBVarType.supported_timestamp_types():
+            return "str"
```

Comment:
Is timestamp type column provided as string type to python udf?

Reply:
That's a good question. I thought we store the timestamp as normal string in offline store table.

Reply:
Let's keep it as string now. Will update integration test & fix it later if it errs.